storage/mysql/log_storage.go

```diff
@@ -489,6 +502,52 @@ func (t *logTreeTX) QueueLeaves(ctx context.Context, leaves []*trillian.LogLeaf,
 	return existingLeaves, nil
 }
 
+func (t *logTreeTX) AddSequencedLeaves(ctx context.Context, leaves []*trillian.LogLeaf) ([]*trillian.QueuedLogLeaf, error) {
+	res := make([]*trillian.QueuedLogLeaf, len(leaves))
+	ok := status.New(codes.OK, "OK").Proto()
+
+	// Note: Leaves are sorted by LeafIndex, so no reordering is necessary.
+	for i, leaf := range leaves {
+		// This should fail on insert, but catch it early.
+		if got, want := len(leaf.LeafIdentityHash), t.hashSizeBytes; got != want {
+			return nil, status.Errorf(codes.FailedPrecondition, "leaves[%d] has incorrect hash size %d, want %d", i, got, want)
+		}
+
+		res[i] = &trillian.QueuedLogLeaf{Status: ok}
+
+		// TODO(pavelkalinnikov): Measure latencies.
+		_, err := t.tx.ExecContext(ctx, insertLeafDataSQL,
+			t.treeID, leaf.LeafIdentityHash, leaf.LeafValue, leaf.ExtraData, 0)
+		// Note: QueueTimestamp == 0 because the entry bypasses the queue.
```

Comment:
Will that mess with the integration latency metrics? (The signer sets integration_timestamp and computes the difference for each leaf, which it then add to a histogram - _might_ be useful for seeing how the mirroring etc. is going on fetching leaves vs. integrating them?)

Reply:
Also, I think we should split the metric in two: one for the `LOG`'s merge delay, another for `PREORDERED_LOG`'s integration delay.

Reply:
@AlCutter ping

Reply:
Sorry, I think splitting the metrics is a good idea.
I guess I was suggesting overloading the `QueueTimestamp`, it's still kinda true in that it's queued to be properly integrated into the tree right?

Reply:
Done timestamp saving, TODOed the new metric bit.
Yeah, kinda, but not queued in the sense we are used to, rather added/stored. Leaving it as is for now though.